client/components/payment-method-logos/index.tsx

```diff
@@ -0,0 +1,113 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { Fragment } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import {
+	Visa,
+	MasterCard,
+	Amex,
+	WooPay,
+	ApplePay,
+	GooglePay,
+	Discover,
+	JCB,
+	Afterpay,
+	Affirm,
+	Klarna,
+} from './Icons';
+
+import './style.scss';
+
+const Payments = [
+	{
+		name: 'visa',
+		component: <Visa key="visa" />,
+	},
+	{
+		name: 'mastercard',
+		component: <MasterCard key="mastercard" />,
+	},
+	{
+		name: 'amex',
+		component: <Amex key="amex" />,
+	},
+	{
+		name: 'discover',
+		component: <Discover key="discover" />,
+	},
+	{
+		name: 'woopay',
+		component: <WooPay key="woopay" />,
+	},
+	{
+		name: 'applepay',
+		component: <ApplePay key="applepay" />,
+	},
+	{
+		name: 'googlepay',
+		component: <GooglePay key="googlepay" />,
+	},
+	{
+		name: 'afterpay',
+		component: <Afterpay key="afterpay" />,
+	},
+	{
+		name: 'affirm',
+		component: <Affirm key="affirm" />,
+	},
+	{
+		name: 'klarna',
+		component: <Klarna key="klarna" />,
+	},
+	{
+		name: 'jcb',
+		component: <JCB key="jcb" />,
+	},
+];
+
+export const WooPaymentMethodLogos: React.VFC< {
+	maxNrElements: number;
+} > = ( { maxNrElements = 10 } ) => {
+	let i = 0;
+	let j = 0;
+	return (
+		<>
+			<div className="woocommerce-payments-method-logos">
+				{ Payments.map( ( payment ) => {
+					if ( i >= maxNrElements ) {
```

Comment:
I see `i` used here as an index. We can use the map's index directly ([from MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#index)):

```js
.map( (paymentMethod, i) => {} )
```

That being said, I think we can change the logic here a bit:

```js
PaymentMethods.slice(0, maxNrElements).map(pm => pm.component);
<div className="woocommerce-payments-method-logos_count">
  + { PaymentMethods.length - maxNrElements }
</div>
```

I'll avoid using const numbers (like 21) because I have no idea what that numbers means and I don't see it described in comments...

Reply:
changed in [af009a5](https://github.com/Automattic/woocommerce-payments/pull/8977/commits/af009a5035144213ca7d93bc162bb035dd0a639d). thx